### ISSUE TITLE
Lot of questions

### ISSUE THREAD

**Issue by thisIsNiceName** [deeplearning4j/deeplearning4j#1560]
Hey guys, so i have many questions to ask.

First of all : i'm doing image recognition with 2 labels ( apple_pie images and baby_back_ribs images )
My problem is that my program is learning very poorly.

You can find my code here : https://gist.github.com/thisIsNiceName/00b7e0467ce0c2c38a2f7d21006184c8

So, at first whenever i do a model.fit(iterTrain) i have a java head space, so I simply can't give the whole iterator to the model, I have to go through each data one at the time.

I tried to use

**while(iterTrain.hasNext())
{
    model.fit(iterTrain.next());
}**

but it learned super poorly and slowly ( one iteration per minute )
**Score at iteration 0 is 281.6533953733086
Score at iteration 0 is 0.022100979709625247
Score at iteration 0 is 0.025564359557628636
etc**

So I tried to send the data one at the time to the model, using a **List<DataSet>** that will contain every DataSet, merge them into one, normalize it, shuffle it ( **I also don't know what shuffle do, someone told me to use it to " batch the labels " or something, i don't really get what it mean**)

And then i just use that huge DataSet as a list, to send every element to the model.

At this point the program does not crash anymore, but it's not learning at all.
We told me to use a ListDataSetIterator, but i don't really see the purpose of it, since it's kinda the same thing at looping through my huge DataSet with the method asList(); 

I also printed :
**while(iterTrain.hasNext())
{
	System.out.println(iterTrain.next().labelCounts());
}**

but the result where all 
**{0=10.0}
...
{0=10.0}
{0=2.0, 1=8.0}
{1=10.0}
...
{1=10.0}**

this is why someone told me to use the shuffle method or " one of the batch method ", wich i did but that didn't change the result at all

Another question : why do we have to load all the image in the memory at the same time ? It can make a few problems if we have a huge of big pictures no ? 

Thank and sorry for the bad english !
